src/StatusQ/Buttons/StatusButton.qml

```diff
@@ -0,0 +1,165 @@
+        switch(statusButton.type) {
+        case StatusButton.Type.Normal:
+            if (statusButton.enabled) {
+                return sensor.containsMouse ? Theme.palette.primaryColor2
+                                            : Theme.palette.primaryColor3;
+            } else {
+                return Theme.palette.baseColor2;
+            }
+
+        case StatusButton.Type.Danger:
+            if (statusButton.enabled) {
+                return sensor.containsMouse ? Theme.palette.dangerColor2
+                                            : Theme.palette.dangerColor3;
+            } else {
+                return Theme.palette.baseColor2;
+            }
+
+        case StatusButton.Type.WithoutBackground:
+            return sensor.containsMouse ? Theme.palette.primaryColor3 : "transparent"
+
+
+        case StatusButton.Type.DangerWithoutBackground:
+            return sensor.containsMouse ? Theme.palette.dangerColor3 : "transparent"
+
+        }
+    }
+
+    border.color: {
+        switch(statusButton.type) {
+        case StatusButton.Type.Normal:
+        case StatusButton.Type.Danger:
+        case StatusButton.Type.WithoutBackground:
+            return "transparent"
+
+        case StatusButton.Type.DangerWithoutBackground:
+            return sensor.containsMouse ? "transparent"
+                                        : Theme.palette.baseColor2
+
+        }
+    }
+
+
+
+    MouseArea {
+        id: sensor
+        width: label.width + statusButton.leftPadding + statusButton.rightPadding
+        height: label.height + statusButton.topPadding + statusButton.bottomPadding
+
+        cursorShape: Qt.PointingHandCursor
+        hoverEnabled: !loading
+
+
+        onPressed: {
+            if (!loading) {
+                statusButton.pressed(mouse)
+            }
+        }
+
+
+        StatusBaseText {
+            id: label
+            anchors.centerIn: parent
+            font.pixelSize: size === StatusButton.Size.Large ? 15 : 13 // by design
+
+            color: {
+                if (loading) {
+                    return "transparent"
+                }
+
+                switch(statusButton.type) {
+
+                case StatusButton.Type.WithoutBackground:
+                case StatusButton.Type.Normal:
+                    if (statusButton.enabled) {
+                        return Theme.palette.primaryColor1;
+                    } else {
+                        return Theme.palette.baseColor1;
+                    }
+
+
+                case StatusButton.Type.DangerWithoutBackground:
+                case StatusButton.Type.Danger:
+                    if (statusButton.enabled) {
+                        return Theme.palette.dangerColor1;
+                    } else {
+                        return Theme.palette.baseColor1;
+                    }
+
+                }
+            }
+        } // Text
+
+        StatusLoadingIndicator {
+            anchors.centerIn: parent
+            visible: loading
```

Comment:
Should we use a `Loader` and make it `active: loading` instead of just using visibility to render this indicator?